spec/models/principal_spec.rb

```diff
@@ -203,6 +203,25 @@
     end
   end
 
+  describe '#onboarded?' do
+    context 'when a principal has been onboarded' do
+      before :each do
+        principal.firm.update_column(:email_address, 'acme@example.com')
```

Comment:
I'll try and wrap this up in a factory. A factory isn't really the best place for encapsulating logic however it will be reusable.

Re: email address:
Currently the logic is if the firm is registered AND if they are remote 
OR
firm is registered AND has an adviser.

For the firm to be registered they need to have been valid at the time of registering.

re: tests
If we were to mock a call on the same object that'd be a smell that we need to change the design and push the :next_onboarded_action somewhere else. Not sure where though. 
Given the amount of setup I'm happy with it living on the principal and the :onboarded? tests unless there is somewhere clearer for the logic to live?